src/libraries/Common/src/System/Security/Cryptography/SlhDsa.cs

```diff
@@ -1114,109 +1562,185 @@ protected virtual void Dispose(bool disposing)
 
         private AsnWriter ExportSubjectPublicKeyInfoCore()
         {
-            byte[] rented = CryptoPool.Rent(Algorithm.PublicKeySizeInBytes);
+            // Public key size for SLH-DSA is at most 64 bytes so we can stack allocate it.
+            int publicKeySizeInBytes = Algorithm.PublicKeySizeInBytes;
+            Debug.Assert(publicKeySizeInBytes is <= 64);
+            Span<byte> publicKey = (stackalloc byte[64])[..publicKeySizeInBytes];
 
-            try
-            {
-                Span<byte> keySpan = rented.AsSpan(0, Algorithm.PublicKeySizeInBytes);
-                ExportSlhDsaPublicKey(keySpan);
+            ExportSlhDsaPublicKeyCore(publicKey);
 
-                AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+            // The ASN.1 overhead of a SubjectPublicKeyInfo encoding a public key is 18 bytes.
+            // Round it off to 32. This checked operation should never throw because the inputs are not
+            // user provided.
+            int capacity = checked(32 + publicKeySizeInBytes);
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER, capacity);
 
+            using (writer.PushSequence())
+            {
                 using (writer.PushSequence())
                 {
-                    using (writer.PushSequence())
-                    {
-                        writer.WriteObjectIdentifier(Algorithm.Oid);
-                    }
+                    writer.WriteObjectIdentifier(Algorithm.Oid);
+                }
+
+                writer.WriteBitString(publicKey);
+            }
+
+            return writer;
```

Comment:
Since we're pre-sizing the AsnWriter, it'd be good to know we're pre-sizing it right.  Adding an assert here like `Debug.Assert(writer.GetEncodedLength() <= capacity);` seems like goodness.